src/pages/workspaces/workspace/Data.js

```diff
@@ -702,17 +704,18 @@ class EntitiesContent extends Component {
   render() {
     const {
       workspace, workspace: { workspace: { namespace, name, attributes: { 'workspace-column-defaults': columnDefaults } }, workspaceSubmissionStats: { runningSubmissionsCount } },
-      entityKey, entityMetadata, loadMetadata, firstRender
+      entityKey, entityMetadata, snapshotDetails, loadMetadata, firstRender
     } = this.props
     const { selectedEntities, deletingEntities, copyingEntities, refreshKey, showToolSelector, igvData: { selectedFiles, refGenome } } = this.state
 
     const { initialX, initialY } = firstRender ? StateHistory.get() : {}
     const selectedKeys = _.keys(selectedEntities)
     const selectedLength = selectedKeys.length
 
-    return selectedFiles ?
-      h(IGVBrowser, { selectedFiles, refGenome, workspace, onDismiss: () => this.setState(_.set(['igvData', 'selectedFiles'], undefined)) }) :
-      h(Fragment, [
+    return Utils.cond(
+      [_.isArray(entityKey), () => snapshotDetails === undefined ? spinnerOverlay : h(SnapshotInfo, { snapshotId: snapshotDetails[entityKey[0]].id, snapshotName: entityKey[0] })],
```

Comment:
Since this code path is a completely separate case from the existing rawls entities, it might be cleaner to branch off one level up, e.g. have `selectionType` return a different token to represent snapshots, and then add a case to the `switchCase` in `render` that could go straight to the right content.